initfiles/examples/EsdlExample/CppPlugin/README.md

```diff
@@ -0,0 +1,82 @@
+    limitations under the License.
+```
+
+# Overview
+
+This folder has an example showing the use of an ESDL service implemented as a C++ plugin to the ESP.
+
+# Environment Setup
+
+This example requires some prior setup of your HPCC Systems deployment. The stock configuration from a platform install has all the HPCC Systems components required. In addition you should have the prerequisites required to build HPCC, outlined at https://github.com/hpcc-systems/HPCC-Platform/wiki/Building-HPCC
+
+## DynamicESDL
+
+You must have configured at least one DynamicESDL ESP Service in this environment, either through configmgr or by editing environment.xml. The instance can be bound on port 0, or on the same port as where you're planning to run your service, or both. The purpose of the instance is to provide the configuration needed for your service. If there's an instance configured on the same port, that instance's configuration will be used, otherwise the configuration of the instance on port 0 will be used. This document assumes you're planning to bind your service to esp process "myesp" and port 8088.
+
+## EclWatch
+
+An instance of EclWatch must be running, and you must have shell access to that node with sudo privilege. This node must also contain two directories for the c++ source compile:
+
+1. Copy of HPCCSystems source- example path is `/home/yourusername/src/hpccsystems`
+2. Location of HPCCSystems build artifacts- example path is `/home/yourusername/builds/hpccsystems`
+
+# C++ Example
+
+Make a copy of the `/opt/HPCCSystems/examples/EsdlExample` folder onto the node running the EclWatch server. Change to the newly copied `EsdlExample/CppPlugin` folder and run the following commands:
+
+1. Generate C++ stub files and base class files, put into a folder named cpptest
+```
+esdl cpp esdl_example.esdl EsdlExample --outdir cpptest
+```
+
+2. Copy the pre-written implementation file over the generated stub in `cpptest/source`
+```
+cp EsdlExampleService.cpp cpptest/source
+```
+
+3. Change to build diretory
+```
+cd cpptest/build
+```
+
+4. Run cmake to create makefiles for the project
+```
+cmake ../source/ -DHPCC_SOURCE_DIR=/home/yourusername/src/hpccsystems -DHPCC_BUILD_DIR=/home/yourusername/builds/hpccsystems -DCMAKE_BUILD_TYPE=Debug -G "Eclipse CDT4 - Unix Makefiles"
+```
+
+5. Compile. The plugin shared object will be created in the `cpptest/build` directory
+```
+make -j8
+```
+
+6. Copy the plugin `libEsdlExampleService.so` to the *plugins* folder under your HPCC running directory. In a stock install the plugins folder is `/opt/HPCCSystems/plugins`
+
+7. Publish the ESDL definition
+```
+esdl publish ../../esdl_example.esdl EsdlExample --overwrite
+```
+
+8. Bind service with the config similar to the following
+```
+esdl bind-service myesp 8088 esdlexample.1 EsdlExample --config ../../cpp_binding.xml --overwrite
+```
+
+9. Test calling the roxie service
+```
+soapplus -url http://.:8088/EsdlExample -i cpprequest.xml
```

Comment:
like above, should be: "../../cpprequest.xml"

Reply:
ok fixed